lib/net.js

```diff
@@ -906,8 +906,11 @@ Socket.prototype.connect = function(options, cb) {
     if (localPort && !util.isNumber(localPort))
       throw new TypeError('localPort should be a number: ' + localPort);
 
-    if (port <= 0 || port > 65535)
-      throw new RangeError('port should be > 0 and < 65536: ' + port);
+    if (options.port !== undefined && parseInt(options.port, 10) !== port)
```

Comment:
Using `parseInt` we would accept `"42foo"` as a valid port, which we wouldn't accept before this change. I don't think we should accept these values, what do you think?